compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -928,7 +928,11 @@ object SpaceEngine {
             && isSubspace(covered, prev)
         then {
           val nullOnly = isNullable && i == len - 1 && isWildcardArg(pat)
-          val msg = if nullOnly then MatchCaseOnlyNullWarning() else MatchCaseUnreachable()
+          val wildcardNotNullable = i == len - 1 && isWildcardArg(pat)
```

Comment:
I don't think this will work. It will catch other cases which are not related to `null`. You may want to modify `isNullable`, since a flexible type is considered "nullable" in terms of values.